### PR TITLE
Added examples files from a slocum and seaglider

### PR DESCRIPTION
**Pull request by emmerbodc**
Files generated from BODC processing chain

### CODE REVIEW

**Comment by callumrollo** [comment]
These are pretty huge, I'd rather not add 40 MB netCDF files to git. Would it be possible to make a new PR with subset files of just a few samples? That will also make it possible to review the .cdl files in the browser, speeding up the review process and future revisions

**Comment by emmerbodc** [comment]
@callumrollo  ok i will look how to easily drop off the data to a few cycles to make it smaller

**Comment by callumrollo** [comment]
contributor_role_vocabulary = "BODC person database"

Should this link to a vocabulary? Perhaps https://vocab.nerc.ac.uk/collection/C86

**Review by callumrollo — CHANGES_REQUESTED** [review]
contributor_role_vocabulary = "BODC person database"

Should this link to a vocabulary? Perhaps https://vocab.nerc.ac.uk/collection/C86

**Comment by emmerbodc** [comment]
added new files @callumrollo @vturpin

**Comment by callumrollo** [comment]
Some confusion in the QC.:

The global attrs include
:rtqc_method = "Raw data, no QC applied." ;
:rtqc_method_doi = "Raw data, no QC applied." ;

But variables have corresponding <PARAM>_QC caraibles

<PARAM>_QC attributes lack flag_values and flag_meanings.

File: og_format_examples_files/sg558_20240206T000000_R.cdl [hunk 1]
```diff
@@ -0,0 +1,549 @@
...
+		:ctd_make_model = "" ;
+		:product_version = "" ;
+		:program = "" ;
+		:network = "" ;
+		:site_vocabulary = "" ;
+		:site = "" ;
+		:keywords_vocabulary = "" ;
+		:web_link = "" ;
+		:title = "OceanGliders trajectory file" ;
+		:platform = "sub-surface gliders" ;
```
Comment (callumrollo) [inline og_format_examples_files/sg558_20240206T000000_R.cdl:389]: there are a few differences between the cdl and the nc. While the cdl has the correct platform:  "sub-surface gliders", the nc has platform: "Autonomous Underwater Vehicle". The agency_role_vocabulary is also difference between .nc and .cdl.

File: og_format_examples_files/sg558_20240206T000000_R.cdl [hunk 2]
```diff
@@ -0,0 +1,549 @@
...
+		:geospatial_lon_min = -53.3449f ;
+		:geospatial_lon_max = -51.54048f ;
+		:geospatial_bounds = "POLYGON ((-53.34490203857422, -51.54047775268555, -63.72288131713867, -63.34503936767578))" ;
+		:geospatial_vertical_min = -0.1495176f ;
+		:geospatial_vertical_max = 751.8447f ;
+		:geospatial_vertical_positive = "down" ;
+		:geospatial_vertical_units = "m" ;
+		:time_coverage_start = "2024-02-06T17:05Z" ;
+		:time_coverage_end = "2024-02-23T20:50Z" ;
+		:agency = "UEA" ;
```
Comment (callumrollo) [inline og_format_examples_files/sg558_20240206T000000_R.cdl:354]: Victor's latest PR change from using "agency" to "institution" throughout these attrs. I need to update the other example cdls too, so we can ignore this for now

File: og_format_examples_files/sg558_20240206T000000_R.cdl [hunk 3]
```diff
@@ -0,0 +1,549 @@
...
+		:agency_role = "contact point" ;
+		:agency_role_vocabulary = "https://edmo.seadatanet.org/" ;
+		:agency_id = "" ;
+		:agency_id_vocabulary = "EDMO" ;
+		:contributor_name = "Karen Heywood" ;
+		:contributor_email = "k.heywood@uea.ac.uk" ;
+		:contributor_role = "Principal investigator" ;
+		:contributor_role_vocabulary = "BODC person database" ;
```
Comment (callumrollo) [inline og_format_examples_files/sg558_20240206T000000_R.cdl:362]: is the BODC person database this vocab linked in the OGformat doc? http://vocab.nerc.ac.uk/collection/W08

Comment (callumrollo) [reply og_format_examples_files/sg558_20240206T000000_R.cdl:362]: Or maybe https://vocab.nerc.ac.uk/collection/C86/current/ ?

File: og_format_examples_files/sg558_20240206T000000_R.cdl [hunk 4]
```diff
@@ -0,0 +1,549 @@
...
+		:creator_url = "https://www.noc.ac.uk" ;
+		:creator_type = "DAC" ;
+		:creator_institution = "NOC" ;
+		:wmoid = "8901054" ;
+		:institution = "" ;
+		:comment = "" ;
+		:history = "" ;
+		:keywords = "" ;
+		:doi = "" ;
+		:ctd_make_model = "" ;
```
Comment (callumrollo) [inline og_format_examples_files/sg558_20240206T000000_R.cdl:380]: quite a few empty attrs that do not feature in the OGformat, like ctd_make_model. Consider removing them?

File: og_format_examples_files/sg558_20240206T000000_R.cdl [hunk 5]
```diff
@@ -0,0 +1,549 @@
...
+		TRAJECTORY:data_mode_vocabulary = "" ;
+	string PLATFORM_TYPE ;
+		PLATFORM_TYPE:long_name = "type of glider" ;
+		PLATFORM_TYPE:platform_type_vocabulary = "" ;
+	string PLATFORM_MODEL ;
+		PLATFORM_MODEL:long_name = "model of the glider" ;
+		PLATFORM_MODEL:platform_model_vocabulary = "" ;
+	string WMO_IDENTIFIER ;
+		WMO_IDENTIFIER:long_name = "wmo id" ;
+	string DEPLOYMENT_TIME ;
```
Comment (callumrollo) [inline og_format_examples_files/sg558_20240206T000000_R.cdl:329]: Format describes DEPLOYMENT_TIME as a double, not string

**Review by callumrollo — CHANGES_REQUESTED** [review]

**Comment by castelao** [comment]
@emmerbodc, good progress! According to the checker, a couple of attributes have the wrong types. Please let me know if you identify any mistakes in the checker recommendations.

**Review by castelao — CHANGES_REQUESTED** [review]
The `valid_min` & `valid_max` should have the same type of the variable.

File: og_format_examples_files/sg558_20240206T000000_R.cdl [hunk 6]
```diff
@@ -0,0 +1,549 @@
...
+		:title = "OceanGliders trajectory file" ;
+		:platform = "sub-surface gliders" ;
+		:platform_vocabulary = "https://vocab.nerc.ac.uk/collection/L06/current/27/" ;
+		:Conventions = "CF-1.8, ACDD-1.3, OG-1.0" ;
+		:naming_authority = "BODC" ;
+		:standard_name_vocabulary = "https://cfconventions.org/Data/cf-standard-names/current/build/cf-standard-name-table.html" ;
+		:processing_level = "No quality control applied to the data." ;
+		:featureType = "trajectory" ;
+		:xglider_type = "trajectoryObs" ;
+		:rtqc_method = "Raw data, no QC applied." ;
```
[the remark this answers is not in the document]
Comment (emmerbodc) [reply og_format_examples_files/sg558_20240206T000000_R.cdl:397]: I think it's useful to show how to display the QC variables even if empty in this example.

File: og_format_examples_files/sg558_20240206T000000_R.cdl (hunk 1 above)
Comment (emmerbodc) [reply og_format_examples_files/sg558_20240206T000000_R.cdl:389]: Yes I manually edited to CDL as our processing chain needs some development work to correct these. I will produce the nc file from the CDL so they stay consistent

File: og_format_examples_files/sg558_20240206T000000_R.cdl (hunk 3 above)
Comment (emmerbodc) [reply og_format_examples_files/sg558_20240206T000000_R.cdl:362]: I can get this updated to the C86 vocab for PI.

File: og_format_examples_files/sg558_20240206T000000_R.cdl (hunk 4 above)
Comment (emmerbodc) [reply og_format_examples_files/sg558_20240206T000000_R.cdl:380]: Yep will remove these legacy attrs. doi i believe is still in the manual as a global attribute - probably should keep this even if empty.

File: og_format_examples_files/sg558_20240206T000000_R.cdl (hunk 5 above)
Comment (emmerbodc) [reply og_format_examples_files/sg558_20240206T000000_R.cdl:329]: ah yes good spot. I think we had this as a string to keep human readable more like a metadata field than a data field but it is easier for a machine to work with.

File: og_format_examples_files/sg558_20240206T000000_R.cdl (hunk 2 above)
Comment (emmerbodc) [reply og_format_examples_files/sg558_20240206T000000_R.cdl:354]: ok will update for these examples

**Comment by emmerbodc** [comment]
> @emmerbodc, good progress! According to the checker, a couple of attributes have the wrong types. Please let me know if you identify any mistakes in the checker recommendations.

@castelao Thanks i haven't run it through the checker, when i try to access it the link isn't working for me. Can you please expand on which attributes have the wrong types please?

**Comment by emmerbodc** [comment]
@callumrollo @castelao @vturpin @justinbuck updated files now uploaded

**Review by callumrollo — APPROVED** [review]

**Comment by castelao** [comment]
> > @emmerbodc, good progress! According to the checker, a couple of attributes have the wrong types. Please let me know if you identify any mistakes in the checker recommendations.
> 
> @castelao Thanks i haven't run it through the checker, when i try to access it the link isn't working for me. Can you please expand on which attributes have the wrong types please?

Hi @emmerbodc , I'm sorry to hear that the checker didn't work for you and I'm interested to understand why it didn't work. Please, send me an email with more information.

Some interesting ones are:
- Lat & Lon variables (including _GPS ones) should be type double (f64). I believe f32 could be an issue for longitude only, but to keep consistency it would make sense to have latitudes f64 as well.
- There are some missing standard names
- One interesting question. Should long_name be a free text or regulated? I see the value of leaving as a free text. What do you all think about it? Right now the checker is considering a fixed text.

**Review by vturpin — APPROVED** [review]

**Comment by emmerbodc** [comment]
@callumrollo  do you mind resolving the merging block here please.

**Comment by callumrollo** [comment]
There's no merge conflicts that I can see, it's blocked from merging by @castelao's review requesting changes

**Review by glidermann — APPROVED** [review]

**Review by castelao — APPROVED** [review]
I don't want to hold you, but I think there are still some issues here.

**Comment by vturpin** [comment]
Thanks @castelao for not holding this. 

```
_Some interesting ones are:

- Lat & Lon variables (including _GPS ones) should be type double (f64). I believe f32 could be an issue for longitude only, but to keep consistency it would make sense to have latitudes f64 as well.
- There are some missing standard names
- One interesting question. Should long_name be a free text or regulated? I see the value of leaving as a free text. What do you all think about it? Right now the checker is considering a fixed text._
```

My opinion is that long name should be regulated. I understand long name as a sort of definition of the variable, and we want DAC to understand the variables similarly.  Leaving it free text will lead to empty long name in too many data sets. That is also a risk I think.

Can someone address the other points ? 
This particular issue can wait a bit to be merge I think. I fear if we do not fix issues now, it will remain erroneous for a long time if we merge now. Unless we open a new issues with the remaining point raised by Gui.
   
@castelao is there other point you see that should be addressed in this file?

**Comment by castelao** [comment]
I ran it through the checker and sent some suggestions directly to @emmerbodc .

Using regulated `long_name` would restrict future use. I think the attribute that you're looking for is the `standard_name`.

**Comment by vturpin** [comment]
I tottally understand this point @castelao . 

Then shall we had an attribute `standard_name` to each variables ? I am in favor of this idea and happy to PR. 
But I would appreciate the view of others.

The PR will simply add  `standard_name` in the variable attribute and copy the fixed value from `long_name` attribute into `standard_name` attribute.

**Comment by emmerbodc** [comment]
Hi @castelao thank you for taking a look at this and providing feedback. I will take a look shortly at the CDL more closely. 

@vturpin @castelao Regarding the long name, the long name is the definition provided in the OG1 vocabulary e.g. 
https://vocab.nerc.ac.uk/collection/OG1/current/
OG concept 
PRES 
:long_name = "Pressure (spatial coordinate) exerted by the water body by profiling pressure sensor and correction to read zero at sea level";
This is the same definition as provided on https://vocab.nerc.ac.uk/collection/OG1/current/PRES/

The definition is important for providing users an understanding of what the measurement is beyond the titled variable.  

Climate and Forecast Standard Names is from a different vocabulary so we don't want to confuse the two. E.g standard name vocab https://vocab.nerc.ac.uk/collection/P07/current/ 
But happy for it to go in attribute standard name.

**Comment by castelao** [comment]
> The PR will simply add `standard_name` in the variable attribute and copy the fixed value from `long_name` attribute into `standard_name` attribute.

No, that would break with `CF`. The `standard_name` is controlled, and using it allows easy integration with other datasets beyond gliders. If `P07` is not associated with CF's `standard_name`, I suggest using both. And I appreciate `P07` defining 'Preferred Label', guiding its use while allowing future flexibility. I think @emmerbodc 's second example is the way to go (I just wish the variables labels were lowercase :) ).

https://cfconventions.org/Data/cf-standard-names/current/build/cf-standard-name-table.html

**Pull request merged**